core/com.b2international.snowowl.datastore.server/src/com/b2international/snowowl/datastore/server/DatastoreServerBootstrap.java

```diff
@@ -147,9 +150,26 @@ public void run(SnowOwlConfiguration configuration, Environment env, IProgressMo
 			initializeJobSupport(env, configuration);
 			initializeRepositories(configuration, env);
 			initializeRequestSupport(env, configuration.getModuleConfig(RepositoryConfiguration.class).getNumberOfWorkers());
+			verifyRepositories(env);
 		}
 	}
 	
+	@SuppressWarnings("deprecation")
+	private void verifyRepositories(Environment env) {
+		final DefaultRepositoryManager repositories = (DefaultRepositoryManager) env.service(RepositoryManager.class);
+		Optional<InternalRepository> inConsistentRepository = repositories.repositories()
+			.stream()
+			.filter(repository -> repository instanceof InternalRepository).map(InternalRepository.class::cast)
+			.filter(repository -> !repository.isConsistent(BranchPathUtils.createMainPath())) 
+			/* XXX: or should we check all the branches, not just the main??*/
+			.findAny();
+		
+		if (inConsistentRepository.isPresent()) {
+				LOG.error("Found inconsistent repository: {}, halting boot.", inConsistentRepository.get().getCdoRepository().getRepositoryName());
+				Thread.currentThread().suspend();
```

Comment:
https://jira.ihtsdotools.org/browse/APDS-124
> If the commits do not match Log a message at FATAL level and make Snow Owl pause. Do not exit the process as this will cause supervisor to restart it.


we may want to do here differently then the original request, that explicitly requested for suspending, rather than halting / deactivating.